presto-main/src/test/java/com/facebook/presto/execution/TestCreateTableTask.java

```
@@ -0,0 +1,181 @@
+import static org.testng.Assert.fail;
+
+public class TestCreateTableTask
+{
+    private static final String CATALOG_NAME = "catalog";
+    private CatalogManager catalogManager;
+    private TypeManager typeManager;
+    private TransactionManager transactionManager;
+    private TablePropertyManager tablePropertyManager;
+    private Catalog testCatalog;
+    private Session testSession;
+    private MockMetadata metadata;
+
+    @BeforeMethod
+    public void setUp()
+            throws Exception
+    {
+        catalogManager = new CatalogManager();
+        typeManager = new TypeRegistry();
+        transactionManager = createTestTransactionManager(catalogManager);
+        tablePropertyManager = new TablePropertyManager();
+        testCatalog = createBogusTestingCatalog(CATALOG_NAME);
+        catalogManager.registerCatalog(testCatalog);
+        tablePropertyManager.addProperties(testCatalog.getConnectorId(),
+                ImmutableList.of(stringSessionProperty("baz", "test property", null, false)));
+        testSession = testSessionBuilder()
+                .setCatalog(CATALOG_NAME)
+                .setSchema(TINY_SCHEMA_NAME)
+                .setTransactionId(transactionManager.beginTransaction(false))
+                .build();
+        metadata = new MockMetadata(typeManager,
+                tablePropertyManager,
+                testCatalog.getConnectorId());
+    }
+
+    @Test
+    public void testCreateTableNotExistsTrue()
+            throws Exception
+    {
+        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
+                ImmutableList.of(new ColumnDefinition("a", "BIGINT", Optional.empty())),
+                true,
+                ImmutableMap.of(),
+                Optional.empty());
+
+        try {
+            getFutureValue(new CreateTableTask().internalExecute(statement, metadata, new AllowAllAccessControl(), testSession, emptyList()));
+        }
+        catch (RuntimeException e) {
```

Remove the catch block and let the exception propagate normally (which fails the test)